src/Symfony/Component/Workflow/Registry.php

```diff
@@ -28,8 +29,16 @@ class Registry
      */
     public function add(Workflow $workflow, $supportStrategy)
     {
-        if (!$supportStrategy instanceof SupportStrategyInterface) {
-            throw new \InvalidArgumentException('The "supportStrategy" is not an instance of SupportStrategyInterface.');
+        @trigger_error(sprintf('%s is deprecated since Symfony 4.0. Use addWorkflow instead.', __METHOD__), E_USER_DEPRECATED);
+        $this->addWorkflow($workflow, $supportStrategy);
+    }
+
+    public function addWorkflow(WorkflowInterface $workflow, $supportStrategy)
+    {
+        if ($supportStrategy instanceof SupportStrategyInterface) {
+            @trigger_error(sprintf('Passing a "%s" in "%s" is deprecated since 4.0. Use "%s" instead.', SupportStrategyInterface::class, __METHOD__, SupportStrategyWorkflowInterface::class), E_USER_DEPRECATED);
```

Comment:
4.1

Reply:
Actually, IMO the new method can easily just require the new interface.

Reply:
I guess if we do that we need to tu dupplicate the logic in addWorkflow (there is not much). Maybe we should do that when going to 5.0 and removing this deprecation ?

Reply:
I agree with @xabbuh, if we don't add the typehint now we won't be able to have it before 6.0. 
Removing the deprecation + exception there lets only ` $this->workflows[] = array($workflow, $supportStrategy);` to duplicate, not a big deal IMHO.